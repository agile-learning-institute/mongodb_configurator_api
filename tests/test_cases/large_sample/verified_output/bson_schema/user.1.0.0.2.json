{
  "description": "A user collection for testing the schema system",
  "bsonType": "object",
  "properties": {
    "_id": {
      "description": "The unique identifier for a user",
      "bsonType": "objectId"
    },
    "user_name": {
      "description": "Username",
      "pattern": "^\\S{1,40}$",
      "bsonType": "string"
    },
    "first_name": {
      "description": "Users First Name",
      "pattern": "^\\S{1,40}$",
      "bsonType": "string"
    },
    "last_name": {
      "description": "Users Last Name",
      "pattern": "^\\S{1,40}$",
      "bsonType": "string"
    },
    "status": {
      "description": "document status",
      "bsonType": "string",
      "enum": [
        "draft",
        "active",
        "archived"
      ]
    },
    "last_saved": {
      "description": "The last time this document was saved",
      "bsonType": "object",
      "properties": {
        "from_ip": {
          "description": "Http Request remote IP address",
          "pattern": "^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$",
          "bsonType": "string"
        },
        "by_user": {
          "description": "ID Of User",
          "pattern": "^\\S{1,40}$",
          "bsonType": "string"
        },
        "at_time": {
          "description": "The date-time when last updated",
          "bsonType": "date"
        },
        "correlation_id": {
          "description": "The logging correlation ID of the update transaction",
          "pattern": "^\\S{1,40}$",
          "bsonType": "string"
        }
      },
      "additionalProperties": false,
      "required": [
        "from_ip",
        "by_user",
        "at_time",
        "correlation_id"
      ]
    }
  },
  "required": [
    "_id",
    "user_name",
    "status",
    "last_saved"
  ],
  "additionalProperties": false
}