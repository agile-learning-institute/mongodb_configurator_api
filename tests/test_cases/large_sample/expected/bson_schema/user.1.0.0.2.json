{
  "bsonType": "object",
  "required": ["_id", "user_name", "status", "last_saved"],
  "properties": {
    "_id": {
      "bsonType": "objectId",
      "pattern": "^[0-9a-fA-F]{24}$"
    },
    "user_name": {
      "bsonType": "string",
      "pattern": "^[a-zA-Z0-9_-]{4,40}$"
    },
    "first_name": {
      "bsonType": "string",
      "pattern": "^[a-zA-Z0-9_-]{4,40}$"
    },
    "last_name": {
      "bsonType": "string",
      "pattern": "^[a-zA-Z0-9_-]{4,40}$"
    },
    "status": {
      "bsonType": "string",
      "enum": ["draft", "active", "archived"]
    },
    "last_saved": {
      "bsonType": "object",
      "required": ["from_ip", "by_user", "at_time", "correlation_id"],
      "properties": {
        "from_ip": {
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        },
        "by_user": {
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        },
        "at_time": {
          "bsonType": "date"
        },
        "correlation_id": {
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        }
      }
    }
  }
} 